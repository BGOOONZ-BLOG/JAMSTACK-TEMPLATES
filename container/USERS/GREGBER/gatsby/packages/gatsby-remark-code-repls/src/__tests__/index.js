jest.mock(`fs`, () => {
  return {
    existsSync: jest.fn(),
    readFileSync: jest.fn(),
  }
})

const fs = require(`fs`)
const Remark = require(`remark`)
const plugin = require(`../index`)

const {
  PROTOCOL_BABEL,
  PROTOCOL_CODEPEN,
  PROTOCOL_CODE_SANDBOX,
  PROTOCOL_RAMDA,
} = require(`../constants`)

const REMARK_TESTS = {
  Babel: PROTOCOL_BABEL,
  Codepen: PROTOCOL_CODEPEN,
  CodeSandbox: PROTOCOL_CODE_SANDBOX,
  Ramda: PROTOCOL_RAMDA,
}

const remark = new Remark()
const JSONstringifySpy = jest.spyOn(JSON, `stringify`)

describe(`gatsby-remark-code-repls`, () => {
  beforeEach(() => {
    fs.existsSync.mockReset()
    fs.existsSync.mockReturnValue(true)

    fs.readFileSync.mockReset()
    fs.readFileSync.mockReturnValue(`const foo = "bar";`)
    JSONstringifySpy.mockClear()
  })

  Object.keys(REMARK_TESTS).forEach(name => {
    describe(`${name} remark transform`, () => {
      const protocol = REMARK_TESTS[name]

      it(`generates a link for the specified example file`, async () => {
        const markdownAST = remark.parse(`[](${protocol}file.js)`)

        const transformed = plugin({ markdownAST }, { directory: `examples` })

        expect(transformed).toMatchSnapshot()
      })

      it(`generates a link with the specified target`, async () => {
        const markdownAST = remark.parse(`[](${protocol}file.js)`)

        const transformed = plugin(
          { markdownAST },
          { directory: `examples`, target: `_blank` }
        )

        expect(transformed).toMatchSnapshot()
      })

      it(`generates a link for files in nested directories`, async () => {
        const markdownAST = remark.parse(
          `[](${protocol}path/to/nested/file.js)`
        )

        const transformed = plugin({ markdownAST }, { directory: `examples` })

        expect(transformed).toMatchSnapshot()
      })

      it(`generates a link with the specified default text`, () => {
        const markdownAST = remark.parse(`[](${protocol}file.js)`)

        const transformed = plugin(
          { markdownAST },
          { directory: `examples`, defaultText: `Click me` }
        )

        expect(transformed).toMatchSnapshot()
      })

      it(`generates a link with the specified inline text even if default text is specified`, () => {
        const markdownAST = remark.parse(
          `[Custom link text](${protocol}file.js)`
        )

        const transformed = plugin(
          { markdownAST },
          { defaultText: `Click me`, directory: `examples` }
        )

        expect(transformed).toMatchSnapshot()
      })

      it(`verifies example files relative to the specified directory`, () => {
        const markdownAST = remark.parse(
          `[](${protocol}path/to/nested/file.js)`
        )

        plugin({ markdownAST }, { directory: `examples` })

        expect(fs.existsSync).toHaveBeenCalledWith(
          `examples/path/to/nested/file.js`
        )
      })

      it(`errors if you do not provide a directory parameter`, () => {
        const markdownAST = remark.parse(`[](${protocol}file.js)`)

        expect(() => plugin({ markdownAST })).toThrow(
          `Required REPL option "directory" not specified`
        )
      })

      it(`errors if you provide an invalid directory parameter`, () => {
        fs.existsSync.mockReturnValue(false)

        const markdownAST = remark.parse(`[](${protocol}file.js)`)

        expect(() => plugin({ markdownAST }, { directory: `fake` })).toThrow(
          `Invalid REPL directory specified "fake"`
        )
      })

      it(`errors if you provide multiple files in non-codesandbox examples`, () => {
        const markdownAST = remark.parse(
          `[](${protocol}path/to/nested/file.js,path/to/nested/anotherFile.js,path/to/nested/file.css)`
        )
        const runPlugin = () =>
          plugin({ markdownAST }, { directory: `examples` })

        if (protocol !== PROTOCOL_CODE_SANDBOX) {
          expect(runPlugin).toThrow(
            `Code example path should only contain a single file, but found more than one: ` +
              `path/to/nested/file.js,path/to/nested/anotherFile.js,path/to/nested/file.css. ` +
              `Only CodeSandbox REPL supports multiple files entries, the protocol prefix of which starts with codesandbox://`
          )
        } else {
          expect(runPlugin).not.toThrow()
        }
      })

      it(`supports includeMatchingCSS`, () => {
        const markdownAST = remark.parse(
          `[](${protocol}path/to/nested/file.js)`
        )
        const runPlugin = () =>
          plugin(
            { markdownAST },
            {
              directory: `examples`,
              codepen: {
                includeMatchingCSS: true,
              },
            }
          )
        expect(runPlugin).not.toThrow()
      })

      if (protocol === PROTOCOL_CODE_SANDBOX) {
        it(`supports custom html config option for index html`, () => {
          const markdownAST = remark.parse(
            `[](${protocol}path/to/nested/file.js)`
          )

          const transformed = plugin(
            { markdownAST },
            {
              directory: `examples`,
              codesandbox: {
                html: `<span id="foo"></span>`,
              },
            }
          )

          expect(transformed).toMatchSnapshot()
        })

        it(`supports custom dependencies config option for NPM module dependencies`, () => {
          const markdownAST = remark.parse(
            `[](${protocol}path/to/nested/file.js)`
          )

          const transformed = plugin(
            { markdownAST },
            {
              codesandbox: {
                dependencies: [
                  `react`,
                  `react-dom@next`,
                  `prop-types@15.5`,
                  `@babel/core@7.4.0`,
                ],
              },
              directory: `examples`,
            }
          )

          expect(JSONstringifySpy).toHaveBeenCalledWith(
            expect.objectContaining({
              files: expect.objectContaining({
                "package.json": expect.objectContaining({
                  content: expect.objectContaining({
                    dependencies: expect.objectContaining({
                      react: `latest`,
                      "react-dom": `next`,
                      "prop-types": `15.5`,
                      "@babel/core": `7.4.0`,
                    }),
                  }),
                }),
              }),
            })
          )
          expect(transformed).toMatchSnapshot()
        })

        it(`supports importing multiple files`, () => {
          const markdownAST = remark.parse(
            `[](${protocol}path/to/nested/file.js,path/to/nested/anotherFile.js,path/to/nested/file.css)`
          )
          const transformed = plugin({ markdownAST }, { directory: `examples` })
          expect(transformed).toMatchSnapshot()
        })
      }
    })
  })
})
