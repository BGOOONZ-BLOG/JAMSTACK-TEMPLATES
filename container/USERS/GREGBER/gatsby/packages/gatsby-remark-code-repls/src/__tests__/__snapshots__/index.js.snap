// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-code-repls Babel remark transform generates a link for files in nested directories 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 35,
          "line": 1,
          "offset": 34,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 35,
      "line": 1,
      "offset": 34,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel remark transform generates a link for the specified example file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel remark transform generates a link with the specified default text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >Click me</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel remark transform generates a link with the specified inline text even if default text is specified 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" >Custom link text</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 36,
          "line": 1,
          "offset": 35,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 36,
      "line": 1,
      "offset": 35,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Babel remark transform generates a link with the specified target 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://babeljs.io/repl/#?presets=react&code_lz=MYewdgzgLgBAZiEMC8MBEAjAhgJzQbiA\\" target=\\"_blank\\" rel=\\"noreferrer\\">REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform generates a link for files in nested directories 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqIQ6WgAsA9LVLzy8ei3mRYxBPyEgJbAB4FZtEVGY1GDJlIA8LCADcABBBYBeADogATqSktL4AfPbyTs4hIPoyCkoqajAaWoQkVnS2SCDWcLSuYIGunq6-AEZofr4A3DF6QA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 41,
          "line": 1,
          "offset": 40,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 41,
      "line": 1,
      "offset": 40,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform generates a link for the specified example file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqJw0QiRD8hICWwAeBABa0RUZjUYMmUgDwsIANwAEEFgF4AOiABOpUrQcA-IwHpTZ9_OFIWGIEZBA9ekYkcIo4WgswFwsbCwcAIzRHBwBueQUgA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 26,
          "line": 1,
          "offset": 25,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 26,
      "line": 1,
      "offset": 25,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform generates a link with the specified default text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqJw0QiRD8hICWwAeBABa0RUZjUYMmUgDwsIANwAEEFgF4AOiABOpUrQcA-IwHpTZ9_OFIWGIEZBA9ekYkcIo4WgswFwsbCwcAIzRHBwBueQUgA\\" >Click me</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 26,
          "line": 1,
          "offset": 25,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 26,
      "line": 1,
      "offset": 25,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform generates a link with the specified inline text even if default text is specified 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqJw0QiRD8hICWwAeBABa0RUZjUYMmUgDwsIANwAEEFgF4AOiABOpUrQcA-IwHpTZ9_OFIWGIEZBA9ekYkcIo4WgswFwsbCwcAIzRHBwBueQUgA\\" >Custom link text</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 42,
          "line": 1,
          "offset": 41,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 42,
      "line": 1,
      "offset": 41,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform generates a link with the specified target 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqJw0QiRD8hICWwAeBABa0RUZjUYMmUgDwsIANwAEEFgF4AOiABOpUrQcA-IwHpTZ9_OFIWGIEZBA9ekYkcIo4WgswFwsbCwcAIzRHBwBueQUgA\\" target=\\"_blank\\" rel=\\"noreferrer\\">REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 26,
          "line": 1,
          "offset": 25,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 26,
      "line": 1,
      "offset": 25,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform supports custom dependencies config option for NPM module dependencies 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTzMBOMTE0QgoaenCYAaEIOEBaFqQC2SEORgAPGSBT9SKBbQCeHBKICMAVgLWQsgAIAjNM5hQA9DUHqA7AQALAQADCAAvrIqaBCUoui0ABaetKSemlIwLJ6QsMQI4ZEgsWxaBIm0KlDMNIwMIiAAPCwQAG4ABBAsALwAOnKkpLT9AHyNni2tIxGyCcmp6fD02bmEJDV09eq1Uu1gg-3d7f2u_P0A3BGFQA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 41,
          "line": 1,
          "offset": 40,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 41,
      "line": 1,
      "offset": 40,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform supports custom html config option for index html 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqIQ6WgAsA9LVLzy8ei3mRYxBPyEgJbAB4FZtEVGY1GDJlIA8cdOQAEEFgF4AOuFKkfAHz28k5o5AEg-jIKSipqMBpahCRWdLZIINZwtC5gfi4eLj4ARmgATj4A3JF6QA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 41,
          "line": 1,
          "offset": 40,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 41,
      "line": 1,
      "offset": 40,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls CodeSandbox remark transform supports importing multiple files 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"https://codesandbox.io/api/v1/sandboxes/define?parameters=N4IgZglgNgpgziAXKADgQwMYGs0HMYB0AVnAPYB2SoGFALjObVSACYwoNvkYTxUC-AGhABbNBEqIQ6WgAsA9LVLzy8ei3mRYxBPyEgJbAB4FZtEVGY1GDJlIA8LCADcABBBYBeADogATqSktL4AfPbyTs4hIPoyCkoqajAaWoQkVnS2SCDWcLSuYIGunq6-AEZofr4A3DHCcYrKqnnJ8mjkQbIwfgBi0GkIyDmZjNm5-YWkxaUgFVUgtbFoco2JLSn9BBhwg9QjdsPkeQVFJeWVNTF6QA\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 95,
          "line": 1,
          "offset": 94,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 95,
      "line": 1,
      "offset": 94,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Codepen remark transform generates a link for files in nested directories 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"/redirect-to-codepen/path/to/nested/file.js\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 37,
          "line": 1,
          "offset": 36,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 37,
      "line": 1,
      "offset": 36,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Codepen remark transform generates a link for the specified example file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"/redirect-to-codepen/file.js\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 22,
      "line": 1,
      "offset": 21,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Codepen remark transform generates a link with the specified default text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"/redirect-to-codepen/file.js\\" >Click me</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 22,
      "line": 1,
      "offset": 21,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Codepen remark transform generates a link with the specified inline text even if default text is specified 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"/redirect-to-codepen/file.js\\" >Custom link text</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 38,
          "line": 1,
          "offset": 37,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 38,
      "line": 1,
      "offset": 37,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Codepen remark transform generates a link with the specified target 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"/redirect-to-codepen/file.js\\" target=\\"_blank\\" rel=\\"noreferrer\\">REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 22,
          "line": 1,
          "offset": 21,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 22,
      "line": 1,
      "offset": 21,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Ramda remark transform generates a link for files in nested directories 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"http://ramdajs.com/repl/#?const%20foo%20%3D%20%22bar%22%3B\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 35,
          "line": 1,
          "offset": 34,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 35,
      "line": 1,
      "offset": 34,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Ramda remark transform generates a link for the specified example file 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"http://ramdajs.com/repl/#?const%20foo%20%3D%20%22bar%22%3B\\" >REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Ramda remark transform generates a link with the specified default text 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"http://ramdajs.com/repl/#?const%20foo%20%3D%20%22bar%22%3B\\" >Click me</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Ramda remark transform generates a link with the specified inline text even if default text is specified 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"http://ramdajs.com/repl/#?const%20foo%20%3D%20%22bar%22%3B\\" >Custom link text</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 36,
          "line": 1,
          "offset": 35,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 36,
      "line": 1,
      "offset": 35,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`gatsby-remark-code-repls Ramda remark transform generates a link with the specified target 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "html",
          "value": "<a href=\\"http://ramdajs.com/repl/#?const%20foo%20%3D%20%22bar%22%3B\\" target=\\"_blank\\" rel=\\"noreferrer\\">REPL</a>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
